src/app/core/core.module.ts

```diff
@@ -122,6 +122,27 @@ import { BrowseDefinition } from './shared/browse-definition.model';
 import { MappedCollectionsReponseParsingService } from './data/mapped-collections-reponse-parsing.service';
 import { ObjectSelectService } from '../shared/object-select/object-select.service';
 
+import {
+  MOCK_RESPONSE_MAP,
+  MockResponseMap,
+  mockResponseMap
+} from './dspace-rest-v2/mocks/mock-response-map';
+import { EndpointMockingRestService } from './dspace-rest-v2/endpoint-mocking-rest.service';
+import { ENV_CONFIG, GLOBAL_CONFIG, GlobalConfig } from '../../config';
+import { SearchFilterService } from './shared/search/search-filter.service';
+import { SearchConfigurationService } from './shared/search/search-configuration.service';
+import { SelectableListService } from '../shared/object-list/selectable-list/selectable-list.service';
+import { RelationshipTypeService } from './data/relationship-type.service';
+import { SidebarService } from '../shared/sidebar/sidebar.service';
+
+export const restServiceFactory = (cfg: GlobalConfig, mocks: MockResponseMap, http: HttpClient) => {
```

Comment:
Could you add a comment around this export to describe it.  Why is this using an `EndpointMockingRestService` when not in production mode?

Reply:
Added some comments for clarification

Reply:
I think there might be a small misunderstanding. This feature doesn't have anything to do with tests. While developing, it often happens a REST endpoint necessary for the frontend is broken, unfinished or returns something unexpected. This service makes it very simple to add a mock REST response instead, so the frontend developer can simply add their expected response there, without having to worry about the actual response from the REST server. 
Of course this should never be used in production, so that's why it's disabled in production mode.

Reply:
So this service allows you to store an example rest response to a json file, and then assign an URL to that json file. While in development mode, all calls to that URL will return the json from the file, all other calls will actually be sent to REST, allowing you to mock endpoints that don't exist or do something unexpected.

Reply:
Thanks for that explanation again. I now see that I had misunderstood the purpose of `MockResponseMap` (which now has better TypeDocs to describe it): https://github.com/DSpace/dspace-angular/blob/b510894f6feddae1eeb0c778d13d423d5a56956f/src/app/shared/mocks/dspace-rest-v2/mocks/mock-response-map.ts

I see now that, by default, this `MockResponseMap` essentially "does nothing".  There are no responses listed by default to mock, so all responses will hit the REST API backend.  However, developers can use `MockResponseMap` to mock a REST response to use for development (as you said, so they can develop on the Angular side even if the REST implementation is not yet complete).

Now that I understand it, I like this idea.  Thanks!